### PR TITLE
Add support of integration tests

### PR DESCRIPTION
**Pull request by extsoft**
- update CI/CD image
- add the ability to work with real git repo
- provide two tests with real git

Affects #89

### CODE REVIEW

File: src/test/addons-git.bash [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+init-repo() {
+    if [ -n "$GIT_REPO_DIR" ]; then
+        _ex mkdir -p $GIT_REPO_DIR
+        _ex cd $GIT_REPO_DIR
+        _ex git init
+        _ex git config --local user.email "elegant-git@example.com"
+        _ex git config --local user.name "Elegant Git"
+        _ex touch $FILE_TO_MODIFY
+        _ex git add .
+        _ex git commit -m "Add $FILE_TO_MODIFY"
```
Comment (AntonKarabaza) [inline src/test/addons-git.bash]: In this case commit message won't be enclosed into double quotes. Better to do `_ex git commit -m \"Add $FILE_TO_MODIFY\"`.

File: src/test/commons.bash [hunk 2]
```diff
@@ -6,8 +6,18 @@ BIN_DIR="$THIS/../../src/main"
 MOCK_DIR="$THIS/../../mock"
 export PATH=$MOCK_DIR:$BIN_DIR:$PATH
 
+check(){
+    run "$@"
+    echo "> Exit code: \$status=$status"
+    for line in ${lines[@]}; do
+        echo "> stdout+stderr: '$line'"
+     done
```
Comment (AntonKarabaza) [inline src/test/commons.bash]: After this loop in output we have each word of executed command output on separate line, not line by line.

File: src/test/addons-git.bash (hunk 1 above)
Comment (extsoft) [reply src/test/addons-git.bash]: Good point

Comment (extsoft) [reply src/test/addons-git.bash]: Fixed

File: src/test/commons.bash (hunk 2 above)
Comment (extsoft) [reply src/test/commons.bash]: Fixed

**Comment by AntonKarabaza** [comment]
@extsoft Nice!

**Pull request merged**